gordo_components/model/models.py

```diff
@@ -283,6 +286,64 @@ def score(
         return explained_variance_score(y, out)
 
 
+class KerasRawModelRegressor(KerasAutoEncoder):
+    """
+    Create a scikit-learn like model with an underlying tensorflow.keras model
+    from a raw config.
+
+    Examples
+    --------
+    >>> import yaml
+    >>> import numpy as np
+    >>> config_str = '''
+    ...   # Arguments to the .compile() method
+    ...   compile:
+    ...     loss: mse
+    ...     optimizer: adam
+    ...
+    ...   # The architecture of the model itself.
+    ...   spec:
+    ...     tensorflow.keras.models.Sequential:
+    ...       layers:
+    ...         - tensorflow.keras.layers.Dense:
+    ...             units: 4
+    ...         - tensorflow.keras.layers.Dense:
+    ...             units: 1
+    ... '''
+    >>> config = yaml.safe_load(config_str)
+    >>> model = KerasRawModelRegressor(kind=config)
+    >>>
+    >>> X, y = np.random.random((10, 4)), np.random.random((10, 1))
+    >>> model.fit(X, y)
+    KerasRawModelRegressor(kind: {'compile': {'loss': 'mse', 'optimizer': 'adam'},
+     'spec': {'tensorflow.keras.models.Sequential': {'layers': [{'tensorflow.keras.layers.Dense': {'units': 4}},
+                                                                {'tensorflow.keras.layers.Dense': {'units': 1}}]}}}
+    )
+    >>> out = model.predict(X)
+    """
+
+    def __init__(self, kind: dict, **kwargs):
+        self.kind = kind  # type: ignore
+        self.kwargs = kwargs
+        self.build_fn = lambda: self.keras_from_spec(self.kind)  # type: ignore
+
+    def __repr__(self):
+        import io
+        from pprint import pprint
+
+        stream = io.StringIO()
+        pprint(self.kind, stream=stream)
+        stream.seek(0)
+        result = f"{self.__class__.__name__}(kind: {stream.read()})"
+        return result
+
+    @staticmethod
+    def keras_from_spec(spec: dict):
+        model = serializer.pipeline_from_definition(spec["spec"])
```

Comment:
Also, what about a quick check if it contains "spec" and "compile" and a nice little exception telling whats wrong if it does not?